tests/main/snap-discard-ns/task.yaml

```diff
@@ -7,17 +7,30 @@ details: |
 prepare: |
     . $TESTSLIB/snaps.sh
     install_local test-snapd-tools
+environment:
+    # TODO: vary by distro later
+    LIBEXECDIR: /usr/lib/snapd/
 execute: |
-    echo "We can discard the namespace before a snap app runs without errors"
-    /usr/lib/snapd/snap-discard-ns test-snapd-tools
+    echo "We can try to discard a namespace before snap runs"
+    $LIBEXECDIR/snap-discard-ns test-snapd-tools
 
-    test-snapd-tools.success
+    echo "We can try to discard a namespace before the .mnt file exits"
+    mkdir -p /run/snapd/ns/
+    $LIBEXECDIR/snap-discard-ns test-snapd-tools
+
+    echo "We can try to discard a namespace before the .mnt file is mounted"
+    touch /run/snapd/ns/test-snapd-tools.mnt
+    $LIBEXECDIR/snap-discard-ns test-snapd-tools
 
-    echo "We can discard the namespace after a snap app runs without errors"
-    /usr/lib/snapd/snap-discard-ns test-snapd-tools
+    echo "We can discard the namespace after a snap runs"
+    test-snapd-tools.success
+    # The last hex is the same as nsfs but older stat on ubuntu 14.04 doesn't know
+    stat -f -c %T /run/snapd/ns/test-snapd-tools.mnt | MATCH 'proc|nsfs|0x6e736673'
```

Comment:
NOTE: This is not a new test, I'm just re-enabling an existing test.

The affected kernel is, for example, one used on CentOS 7. I tested 3.10.x which doesn't have nsfs yet. The output of stat is similar, except for the filesystem magic number.